### PR TITLE
Replace TravisCI with Github Actions

### PR DESCRIPTION
**Pull request by Christopher-Tom**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+#     strategy:
+#       matrix:
+#         python-version: [3.8]
```
Comment (evanugarte) [inline .github/workflows/main.yml:12]:
```suggestion
```

**Review by evanugarte — APPROVED** [review]
[we played this](https://www.youtube.com/watch?v=kUsFWO08CO0) right as it passed

**Review by seemavora — COMMENTED** [review]
remove comments

**Review by seemavora — COMMENTED** [review]
image
I purposely made npm run test fail, but checks are still passing. See Issue #73

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: RPC Tests
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (seemavora) [inline .github/workflows/main.yml:9]: I think Isita mentioned this in the Corev4 File, but is there a reason why it's just Ubuntu?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - run: python3 -m pip install flake8
+      - run: cp client/config/config.example.js client/config/config.js
+      - run: npm install
+      - run: npm run lint
```
Comment (seemavora) [inline .github/workflows/main.yml:26]: Re-tested with proper failure code, and it works. Love the error log much better than travis, looks more cleaner! 
image

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      - run: python3 -m flake8
+      - run: npm run test
```
Comment (seemavora) [inline .github/workflows/main.yml:28]: look at the comments above, but everything else looks good to me!

**Review by seemavora — APPROVED** [review]

**Review by ThaiQ — APPROVED** [review]

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix: 
+       os: [ubuntu-latest, macos-latest, windows-latest]
+             
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:17]: It seems like we don't have a node version in `package.json`? Possibly omit to use whatever version of node is on the OS.

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+name: RPC Tests
+
+on:
+  pull_request:
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:4]: see [this doc](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)

**Review by charlotte-zhuang — CHANGES_REQUESTED** [review]
It looks good. I think you need to change line 4, and we'd want to test a PR as well.

**Comment by seemavora** [comment]
checked it again @Christopher-Tom and it works.

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix: 
+       os: [ubuntu-latest, macos-latest, windows-latest]
+             
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:14]:
```suggestion
      - name: Use Node.js
```

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/setup-node@v1
+      - name: Set up Python ${{ matrix.python-version }}
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:16]:
```suggestion
      - name: Set up Python
```

File: .github/workflows/main.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/setup-python@v2
+        with:
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:18]:
```suggestion
```

File: .github/workflows/main.yml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+          python-version: ${{ matrix.python-version }}
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:19]:
```suggestion
```

**Review by charlotte-zhuang — CHANGES_REQUESTED** [review]
`matrix.python-version` is undefined

File: .github/workflows/main.yml [hunk 11]
```diff
@@ -0,0 +1,23 @@
+name: RPC Tests
+
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
```
Comment (evanugarte) [inline .github/workflows/main.yml:7]: make it just ubuntu

**Review by evanugarte — APPROVED** [review]

**Review by charlotte-zhuang — COMMENTED** [review]
It looks good to me. See Evan's comments too. I wonder if this might have the same issue as Core-v4 where failed tests don't failed the check as well on Github.

**Review by charlotte-zhuang — APPROVED** [review]

**Pull request merged**